### PR TITLE
Handle OSF preprint revisions by using `hypothesis annotation date`

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/725
(also see: https://github.com/elifesciences/enhanced-preprints-import/issues/1150)

### CODE REVIEW

File: data_hub_api/docmaps/v2/sql/docmaps_index.sql [hunk 1]
```diff
@@ -1,7 +1,19 @@
-WITH t_hypothesis_annotation_with_doi AS (
+WITH t_hypothesis_annotation_with_cleaned_uri AS (
   SELECT
-    * EXCEPT (id, created),
-    CASE 
+    * EXCEPT (id, created, uri),
+    annotation.id AS hypothesis_id,
+    annotation.created AS annotation_created_timestamp,
+    DATE(annotation.created) AS annotation_created_date,
+    REGEXP_REPLACE(uri, r'/$', '') AS uri
```
Comment (de-code) [inline data_hub_api/docmaps/v2/sql/docmaps_index.sql:7]: Just a FYI, [RTRIM](https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions#rtrim) can also be used to remove trailing characters (not just space), e.g.:

```sql
RTRIM(uri, '/')
```

File: data_hub_api/docmaps/v2/sql/docmaps_index.sql [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+  FROM `elife-data-pipeline.prod.v_hypothesis_annotation` AS annotation
+  WHERE annotation.group = 'q5X6RWJ6'
+    AND created >= '2022-09-01' -- to ignore any public reviews posted before Sep 2022
+),
+
+t_hypothesis_annotation_with_doi AS (
+  SELECT
+  *,
```
Comment (de-code) [inline data_hub_api/docmaps/v2/sql/docmaps_index.sql]: Internal linter says this should be indented.

File: data_hub_api/docmaps/v2/sql/docmaps_index.sql [hunk 3]
```diff
@@ -127,6 +137,14 @@ t_hypothesis_annotation_with_evaluation_suffix AS (
   FROM t_hypothesis_annotation_with_osf_doi AS annotation
   INNER JOIN t_hypothesis_with_source_doi_rank AS doi_rank
     ON annotation.uri = doi_rank.uri
+    AND (
+      CASE 
+        WHEN annotation.osf_preprint_version_rank IS NOT NULL
```
Comment (de-code) [inline data_hub_api/docmaps/v2/sql/docmaps_index.sql]: For the condition I would find following easier to read:

```sql
annotation.osf_preprint_version_rank IS NULL
OR annotation.osf_preprint_version_rank = doi_rank.source_doi_rank
```

But then somehow it feels "odd" to make the default true. (I will need to think more about it)

Comment (HazalCiplak) [reply data_hub_api/docmaps/v2/sql/docmaps_index.sql]: Done :) "annotation.osf_preprint_version_rank IS NULL" is kind of default true for the not null ones :D

**Review by de-code — APPROVED** [review]
Setting reminder in six month...

**Pull request merged**